en/docs/install-and-setup/setup/security/logins-and-passwords/harshicrop-vault-extension.md

```diff
@@ -0,0 +1,298 @@
+
+# Integrate with HashiCorp Vault
+
+Using [HashiCorp Vault extension](https://github.com/wso2-extensions/carbon-securevault-hashicorp/tree/master), you can set up HashiCorp Vault to store passwords that are mapped to aliases instead of the actual passwords.
+
+## Setting up using root token authentication
+
+### Step 1: Setup HashiCorp Vault
+
+1. Start the HashiCorp Vault server and create a new **kv engine**. Enter a path name when creating the kv engine (e.g., `wso2apim`). The following commands can be used to add secrets with the HashiCorp Vault.
+
+    ```bash
+    # Create a new kv engine
+    vault secrets enable -path=wso2apim -version=2 kv
+
+    # Add new secret
+    vault kv put wso2apim/admin_password value=admin
+
+    # Get a secret (To check)
+    vault kv get -field=value wso2apim/admin_password
+    ```
+
+### Step 2: Configure HashiCorp Vault extension
+
+1. Clone [HashiCorp Vault extension](https://github.com/wso2-extensions/carbon-securevault-hashicorp/tree/master) repo
+
+2. Build the HashiCorp Vault Integration OSGI bundle using `mvn clean install` and copy
+   the `target/org.wso2.carbon.securevault.hashicorp-1.0.jar` file to the `<APIM_HOME>/repository/components/dropin/`
+   directory.
+
+3. Add **HashiCorp Vault Java Driver** (e.g., `vault-java-driver-5.1.0.jar`) to the
+   `<APIM_HOME>/repository/components/lib/` directory.
```

Comment:
```suggestion
   `<API-M_HOME>/repository/components/lib/` directory.
```

Reply:
corrected in 1d8bdffa01d44e5da79703dd2060f0a621bb659a